src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -536,6 +536,54 @@ function double_binary_tree(k::Integer)
     return g
 end
 
+"""
+    regular_tree(k::Integer, z::integer)
+
+Create a [regular tree](https://en.wikipedia.org/wiki/Bethe_lattice), 
+also known as Bethe Lattice or Cayley Tree,
+of depth `k` and degree `z`.
+
+# Examples
+```jldoctest
+julia> regular_tree(4, 3)
+{40, 39} undirected simple Int64 graph
+
+julia> regular_tree(5, 2) == binary_tree(5)
+true
+```
+"""
+function regular_tree(k::T, z::T) where {T<:Integer}
+    k <= 0 && return SimpleGraph(0)
+    k == 1 && return SimpleGraph(1)
+    if Graphs.isbounded(k) && BigInt(z)^k - 1 > typemax(k)
+        throw(DomainError(k, "z^k - 1 not representable by type $T"))
```

Comment:
Here I would change the error from DomainError to InexactError.
I think the error is similar to:
```jl
julia> a::Int8=1923458675849328374548392
ERROR: InexactError: trunc(Int8, 1923458675849328374548392)
```
In our case `a` is `z^k`. What do you think?